src/storm/logic/TimeBoundType.h

```diff
@@ -11,7 +11,7 @@ enum class TimeBoundType { Steps, Time, Reward };
 
 class TimeBoundReference {
     TimeBoundType type;
-    std::string rewardName;
+    boost::optional<std::string> rewardName;
```

Comment:
I think we are trying to move away from boost::optional in favour of std::optional.